client/state/themes/selectors/get-themes-for-query-ignoring-page.js

```diff
@@ -27,13 +28,24 @@ export const getThemesForQueryIgnoringPage = createSelector(
 
 		// If query is default, filter out recommended themes.
 		if ( ! ( query.search || query.filter || query.tier ) ) {
+			const activeThemeId = state.ui
+				? state.themes.activeThemes[ getSelectedSiteId( state ) ]
+				: null;
 			const recommendedThemes = state.themes.recommendedThemes.themes;
-			const themeIds = flatMap( recommendedThemes, ( theme ) => {
-				return theme.id;
-			} );
-			themesForQueryIgnoringPage = themesForQueryIgnoringPage.filter( ( theme ) => {
-				return ! themeIds.includes( theme.id );
-			} );
+			const themeIds = flatMap( recommendedThemes, ( theme ) => theme.id );
+
+			themesForQueryIgnoringPage = themesForQueryIgnoringPage.filter(
+				( theme ) => ! themeIds.includes( theme.id )
+			);
+
+			if ( activeThemeId ) {
+				for ( let i = 0; i < themesForQueryIgnoringPage.length; i++ ) {
+					if ( themesForQueryIgnoringPage[ i ].id === activeThemeId ) {
+						themesForQueryIgnoringPage.unshift( ...themesForQueryIgnoringPage.splice( i, 1 ) );
```

Comment:
Could we select the active theme from the redux store instead of getting it from the theme query result? The reason is if it doesn't show on the first page, then the user won't see it at the beginning

Reply:
Even though the query fetches all themes (see filename), I've refactored the code so that we leverage data store selectors to fetch the active theme. Let me know if this approach works 🙂